website/src/components/McapRecordingDemo/videoCapture.ts

```diff
@@ -66,36 +85,106 @@ export function startVideoCapture(params: VideoCaptureParams): () => void {
   };
 }
 
+/**
+ * Start a periodic capture of camera frames and encode them as H264 video if supported by the
+ * browser, otherwise fall back to JPEG encoding.
+ */
 async function startVideoCaptureAsync(
   params: VideoCaptureParams,
   signal: AbortSignal,
 ) {
-  const { video, onFrame, frameDurationSec } = params;
+  const { video, onJpegFrame, onH264Frame, onError, frameDurationSec } = params;
   const canvas = document.createElement("canvas");
   canvas.width = video.videoWidth;
   canvas.height = video.videoHeight;
   const ctx = canvas.getContext("2d");
 
-  let framePromise: Promise<void> | undefined;
+  let encoder: VideoEncoder | undefined;
+  const framePool: ArrayBuffer[] = [];
+  setupEncoder: try {
+    const config: VideoEncoderConfig = {
+      codec: "avc1.42001f", // Baseline profile (42 00) with maximum level (1f)
+      width: video.videoWidth,
+      height: video.videoHeight,
+      latencyMode: "realtime",
+      avc: { format: "annexb" },
+    };
+    if (typeof VideoEncoder !== "function") {
+      break setupEncoder;
+    }
+    if ((await VideoEncoder.isConfigSupported(config)).supported !== true) {
+      break setupEncoder;
```

Comment:
What is the error message when it breaks like this? Why not throw a more specific error message?

Reply:
It falls back to jpeg encoding

Reply:
Ah thought it still went through the catch